subsys/logging/log_core.c

```diff
@@ -0,0 +1,412 @@
+#define CONFIG_LOG_PRINTK_MAX_STRING_LENGTH 1
+#endif
+
+#ifdef CONFIG_LOG_BACKEND_CONSOLE
+#include <logging/log_backend_console.h>
+LOG_BACKEND_CONSOLE_DEFINE(log_backend_console);
+#endif
+
+static struct log_list_t list;
+
+static bool panic_mode;
+static bool initialized;
+
+static u32_t nullstamp_func(void)
+{
+	return 0;
+}
+
+static timestamp_get_t timestamp_func = nullstamp_func;
+
+
+
+static inline void msg_finalize(struct log_msg *msg,
+				struct log_msg_ids src_level)
+{
+	msg->hdr.ids = src_level;
+	msg->hdr.timestamp = timestamp_func();
+	unsigned int key = irq_lock();
+
+	log_list_add_tail(&list, msg);
+
+	irq_unlock(key);
+
+	if (panic_mode) {
+		(void)log_process();
+	}
+}
+void log_0(const char *str, struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_0(str);
+
+	if (msg == NULL) {
+		return;
+	}
+	msg_finalize(msg, src_level);
+}
+
+void log_1(const char *str,
+	   u32_t arg0,
+	   struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_1(str, arg0);
+
+	if (msg == NULL) {
+		return;
+	}
+	msg_finalize(msg, src_level);
+}
+
+void log_2(const char *str,
+	   u32_t arg0,
+	   u32_t arg1,
+	   struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_2(str, arg0, arg1);
+
+	if (msg == NULL) {
+		return;
+	}
+
+	msg_finalize(msg, src_level);
+}
+
+void log_3(const char *str,
+	   u32_t arg0,
+	   u32_t arg1,
+	   u32_t arg2,
+	   struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_3(str, arg0, arg1, arg2);
+
+	if (msg == NULL) {
+		return;
+	}
+
+	msg_finalize(msg, src_level);
+}
+
+void log_4(const char *str,
+	   u32_t arg0,
+	   u32_t arg1,
+	   u32_t arg2,
+	   u32_t arg3,
+	   struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_4(str, arg0, arg1,
+					       arg2, arg3);
+
+	if (msg == NULL) {
+		return;
+	}
+
+	msg_finalize(msg, src_level);
+}
+
+void log_5(const char *str,
+	   u32_t arg0,
+	   u32_t arg1,
+	   u32_t arg2,
+	   u32_t arg3,
+	   u32_t arg4,
+	   struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_5(str, arg0, arg1,
+					       arg2, arg3, arg4);
+
+	if (msg == NULL) {
+		return;
+	}
+
+	msg_finalize(msg, src_level);
+}
+
+void log_6(const char *str,
+	   u32_t arg0,
+	   u32_t arg1,
+	   u32_t arg2,
+	   u32_t arg3,
+	   u32_t arg4,
+	   u32_t arg5,
+	   struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_create_6(str, arg0, arg1, arg2,
+					       arg3, arg4, arg5);
+
+	if (msg == NULL) {
+		return;
+	}
+
+	msg_finalize(msg, src_level);
+}
+
+void log_hexdump(const u8_t *data,
+		 u32_t length,
+		 struct log_msg_ids src_level)
+{
+	struct log_msg *msg = log_msg_hexdump_create(data, length);
+
+	if (msg == NULL) {
+		return;
+	}
+
+	msg_finalize(msg, src_level);
+}
+
+int log_printk(const char *fmt, va_list ap)
+{
+	if (IS_ENABLED(CONFIG_LOG_PRINTK)) {
+		if (!initialized) {
+			log_init(nullstamp_func, 0);
+		}
+
+		u8_t formatted_str[CONFIG_LOG_PRINTK_MAX_STRING_LENGTH];
```

Comment:
In zephyr we have variables declared in the beginning of the block.

Reply:
ok, i couldn't find it in coding style (http://docs.zephyrproject.org/contribute/contribute_guidelines.html#coding-style). It would be worth to put it somewhere.